x/gasless/client/cli/tx.go

```diff
@@ -0,0 +1,444 @@
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+			feeDenom := args[0]
+			if err := sdk.ValidateDenom(feeDenom); err != nil {
+				return fmt.Errorf("invalid fee denom: %w", err)
+			}
+
+			maxFeeUsagePerTx, ok := sdk.NewIntFromString(args[1])
+			if !ok {
+				return fmt.Errorf("invalid max-fee-usage-per-tx: %s", args[1])
+			}
+
+			maxTxsCountPerConsumer, err := strconv.ParseUint(args[2], 10, 64)
+			if err != nil {
+				return fmt.Errorf("parse max-txs-count-per-consumer: %w", err)
+			}
+
+			maxFeeUsagePerConsumer, ok := sdk.NewIntFromString(args[3])
+			if !ok {
+				return fmt.Errorf("invalid max-fee-usage-per-consumer: %s", args[3])
+			}
+
+			txsAllowed, err := ParseStringSliceFromString(args[4], ",")
+			if err != nil {
+				return err
+			}
+
+			contractsAllowed, err := ParseStringSliceFromString(args[5], ",")
+			if err != nil {
+				return err
+			}
+
+			gasDeposit, err := sdk.ParseCoinNormalized(args[6])
+			if err != nil {
+				return fmt.Errorf("invalid gas-deposit: %w", err)
+			}
+
+			msg := types.NewMsgCreateGasTank(
+				clientCtx.GetFromAddress(),
+				feeDenom,
+				maxFeeUsagePerTx,
+				maxTxsCountPerConsumer,
+				maxFeeUsagePerConsumer,
+				txsAllowed,
+				contractsAllowed,
+				gasDeposit,
+			)
+			if err = msg.ValidateBasic(); err != nil {
+				return err
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+
+	flags.AddTxFlagsToCmd(cmd)
+
+	return cmd
+}
+
+func NewAuthorizeActorsCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "update-authorized-actors [gas-tank-id] [actors]",
+		Args:  cobra.ExactArgs(2),
+		Short: "Update authorized actors of the gas tank",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Update authorized actors of the gas tank.
+Example:
+$ %s tx %s update-authorized-actors 1 aib1...,aib2... --from mykey
+`,
+				version.AppName, types.ModuleName,
+			),
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			gasTankID, err := strconv.ParseUint(args[0], 10, 64)
+			if err != nil {
+				return fmt.Errorf("parse gas-tank-id: %w", err)
+			}
+
+			actors, err := ParseStringSliceFromString(args[1], ",")
+			if err != nil {
+				return err
+			}
+
+			sanitizedActors := []sdk.AccAddress{}
+			for _, actor := range actors {
+				sanitizedActor, err := sdk.AccAddressFromBech32(actor)
+				if err != nil {
+					return err
+				}
+				sanitizedActors = append(sanitizedActors, sanitizedActor)
+			}
+
+			msg := types.NewMsgAuthorizeActors(
+				gasTankID,
+				clientCtx.GetFromAddress(),
+				sanitizedActors,
+			)
+			if err = msg.ValidateBasic(); err != nil {
+				return err
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		},
+	}
+
+	flags.AddTxFlagsToCmd(cmd)
+
+	return cmd
+}
+
+func NewUpdateGasTankStatusCmd() *cobra.Command {
```

Comment:
instead of toggling, I suggest we have two commands: `enable-gas-tank` and `disable-gas-tank`
Toggling will require the caller to first check the status of the gas tank before updating it